flybywiresim/aircraft
# fix: make datastorage aircraft specific

Fixes #[issue_no]

## Summary of Changes


- We need to distinguish between A380 and A320 properties. Using the project-prefix env variable to achieve this.
- Cleanup of some legacy usages of shared code, which was moved to fbw-common (including persistence)



## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub):

## Testing instructions


In the A380X:

 1. Load in the plane, change some configuration options in the EFB. 
     Some examples (best to test as much as possible):
      1. Enter override user for simbrief and import a flight plan
      2. Navigraph login, oans, simbrief, charts, (DM me for build with the secrets enabled if you wish to test this)
      3. Telex, simbridge port and connection, volume sliders, acceleration height...
      4. display self-test time
      5. efis sync
 2. Restart the flight and ensure the properties are saved and have not changed from what you've entered
 3. Perform a flight ideally

In the A320:
 1. Take the examples (and more if you wish) from above and note down the values you have configured BEFORE using this PR
 2. Download the PR and verify the settings did not change
 3. Verify the settings are applied correctly after loading (check efis sync, kg/lbs, accel height for example).
 4. Change some values and verify they are applied properly (like in 3. for example)
 5. Perform a flight ideally


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
